### PR TITLE
Test that various MessageEvents are trusted

### PR DESCRIPTION
**Pull request by domenic**
See https://github.com/whatwg/html/pull/1935.

### CODE REVIEW

**Comment by domenic** [comment]
Note that I cannot get the BroadcastChannel tests to work in Chrome or Firefox---they timeout instead---so if someone can figure out my mistake that would be great.

**Comment by annevk** [comment]
You need to have at least two `BroadcastChannel` objects I think. Ideally in separate globals. (Maybe use an `<iframe>`?)

**Comment by sideshowbarker** [comment]
> > I cannot get the BroadcastChannel tests to work in Chrome or Firefox---they timeout instead

> You need to have at least two `BroadcastChannel` objects I think. Ideally in separate globals.

The following works (though not sure it’s the best way to do it…)

```diff
diff --git a/webmessaging/MessageEvent-trusted.html b/webmessaging/MessageEvent-trusted.html
index e3d26f8..502cd51 100644
--- a/webmessaging/MessageEvent-trusted.html
+++ b/webmessaging/MessageEvent-trusted.html
@@ -1,3 +1,8 @@
+<!--
+const channel = new BroadcastChannel("name");
+channel.postMessage("ping");
+/*
+-->
 <!DOCTYPE html>
 <title>MessagePort message events are trusted</title>
 <script src="/resources/testharness.js"></script>
@@ -26,13 +31,13 @@ async_test(t => {
 async_test(t => {
   assert_true("BroadcastChannel" in self, "The browser must support BroadcastChannel");

+  const worker = new Worker("#");
   const channel = new BroadcastChannel("name");

   channel.onmessage = t.step_func_done(e => {
     assert_equals(e.isTrusted, true);
   });

-  channel.postMessage("ping");
 }, "With a BroadcastChannel");

 async_test(t => {
@@ -43,3 +48,6 @@ async_test(t => {
   window.postMessage("ping", "*");
 }, "With window");
 </script>
+<!--
+*/
+-->
```

(btw would be nice if we could just use template literals in WPT tests but I’m told we shouldn’t yet)

**Comment by domenic** [comment]
Thanks all, fixed the test, and now it indeed passes in Chrome and Firefox.

**Comment by cdumez** [comment]
Wow, it now runs tests in browsers!? so cool!
I wish it would cover Safari (TP) too :D

File: webmessaging/MessageEvent-trusted.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    assert_equals(e.isTrusted, true);
+  });
+
+  channel.port1.postMessage("ping");
+}, "With a MessageChannel and its MessagePorts");
+
+async_test(t => {
+  assert_true("BroadcastChannel" in self, "The browser must support BroadcastChannel");
+
+  new Worker("MessageEvent-trusted-worker.js");
```
Comment (mkruisselbrink) [inline webmessaging/MessageEvent-trusted.html]: It is probably safer to not create this worker until after you've created the BroadcastChannel instance and added the onmessage event handler. I don't think there is anything that guarantees the worker won't be created, ran, and posts its message to the channel before this method has a chance to connect to the channel itself (unless I'm missing something in how cross-thread-synchronization is specified in the html spec)

**Comment by domenic** [comment]
@mkruisselbrink fixed, thanks!

**Comment by domenic** [comment]
Ping @mkruisselbrink for a final sign-off? This seems to have gotten left behind at some point.

**Comment by mkruisselbrink** [comment]
> Ping @mkruisselbrink for a final sign-off? This seems to have gotten left behind at some point.

Looks good to me.

**Pull request merged**